### PR TITLE
StateProof verifier rejects long proofs

### PR DESCRIPTION
**Pull request by algonathan**

### CODE REVIEW

File: crypto/stateproof/verifier.go [hunk 1]
```diff
@@ -65,6 +66,10 @@ func MkVerifierWithLnProvenWeight(partcom crypto.GenericDigest, lnProvenWt uint6
 // Verify checks if s is a valid state proof for the data on a round.
 // it uses the trusted data from the Verifier struct
 func (v *Verifier) Verify(round uint64, data MessageHash, s *StateProof) error {
+	if s.SigProofs.TreeDepth > 20 || s.PartProofs.TreeDepth > 20 {
```
Comment (algoidan) [inline crypto/stateproof/verifier.go:69]: please move the `20` to the const file.

File: crypto/stateproof/verifier_test.go [hunk 2]
```diff
@@ -155,3 +155,27 @@ func TestEqualVerifiers(t *testing.T) {
 
 	a.Equal(verifierLnP, verifier)
 }
+
+func TestTreeDepth(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+
+	p := generateProofForTesting(a, false)
```
Comment (algoidan) [inline crypto/stateproof/verifier_test.go:163]: nits: will it be too hard/takes too much time to actually create a stateproof with >20 tree depth?

File: crypto/stateproof/verifier.go [hunk 3]
```diff
@@ -26,8 +26,9 @@ import (
 
 // Errors for the StateProof verifier
 var (
-	ErrCoinNotInRange = errors.New("coin is not within slot weight range")
-	ErrNoRevealInPos  = errors.New("no reveal for position")
+	ErrCoinNotInRange    = errors.New("coin is not within slot weight range")
+	ErrNoRevealInPos     = errors.New("no reveal for position")
+	ErrTreeDepthTooLarge = errors.New("state proof tree depth is too large")
```
Comment (algoidan) [inline crypto/stateproof/verifier.go:31]: 2 nits:
1- can we make the message more accurate? - i.e use the word `participantTree` and `SignatureTree`
2- should we format this error message with the actual values?

**Review by algoidan — COMMENTED** [review]
LGTM
Had some few Minor nits

File: crypto/stateproof/verifier_test.go (hunk 2 above)
Comment (algonathan) [reply crypto/stateproof/verifier_test.go:163]: I can't make it run on my local machine. Do we need such an excessive test?

**Pull request merged**